### ISSUE TITLE
hlint and ghc-mod doesn't work correctly

### ISSUE THREAD

**Issue by AurevoirXavier** [alanz/vscode-hie-server#81]
It always 'flash'.

![](https://i.imgur.com/veWsuSn.gif)

**Comment by Rhywun** [comment]
I see something similar in which hlint problems appear in the list and then immediately disappear. When I close and re-open the file, the problems are back in the list, until I start typing again and then they disappear.

**Comment by alanz** [comment]
By default we turn off hlint warnings if there are ghc-mod errors, so it does not clutter.

But hlint gives results first, and they get passed through. This gives the flicker, I guess.

**Comment by Rhywun** [comment]
Update: I think I have narrowed down the issue by playing around with VSCode's autosave feature. With autosave of files on, it behaves as shown above. When I turned autosave off, the behavior changed as follows:

1. Open file - hlint problems are highlighted
2. Type some characters in the file - no change, hlint problems are still highlighted
3. Save file - hlint problems disappear
4. Continue typing - hlint problems appear again

So basically, for some reason saving the file is making the hlint problems disappear.

**Comment by alanz** [comment]
Ok, we recently reworked the diagnostics generation, and there *is* a difference between the on-save and on-change behaviour, I will take a look.

**Comment by dnsdhrj** [comment]
Hello,
I found a way to workaround the annoying flickering problem.

1. Go to `Open user settings` in Command Palette (Ctrl+Shift+p).
2. Remove the line  `"languageServerHaskell.hlintOn": true`, if any.
3. Save the setting.
4. Add the line  `"languageServerHaskell.hlintOn": true`.
5. Save the setting.
6. hlint appears!

But, if I type some code and save it, hlint disappears again. (only that file)
So this is just a hotfix for those who desperately needs linting feature now. Maybe this has to do with actual cause?

**Comment by dnsdhrj** [comment]
By the way, this issue seems like a duplicate of #59 .

**Comment by jneira** [comment]
hlint should be fixed in latest hie and hie does not use ghcmod anymore
@AurevoirXavier feel free to reopen if you continue experiencing it with the latest version of hie
